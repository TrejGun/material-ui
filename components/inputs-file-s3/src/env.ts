declare global {
  namespace NodeJS {
    interface ProcessEnv {
      BE_URL: string;
      AWS_S3_BUCKET: string;
      AWS_REGION: string;
    }
  }
}

export {};
