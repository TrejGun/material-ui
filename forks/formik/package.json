{
  "author": "Jared Palmer <jared@palmer.net> (https://jaredpalmer.com)",
  "dependencies": {
    "deepmerge": "2.2.1",
    "hoist-non-react-statics": "3.3.2",
    "lodash": "4.17.21",
    "lodash-es": "4.17.21",
    "react-fast-compare": "2.0.4",
    "tiny-warning": "1.0.3",
    "tslib": "1.14.1"
  },
  "description": "Build forms in React, without the tears",
  "devDependencies": {
    "@testing-library/react": "11.2.7",
    "@types/hoist-non-react-statics": "3.3.1",
    "@types/lodash": "4.14.182",
    "@types/react": "16.14.25",
    "@types/react-dom": "16.9.15",
    "@types/warning": "3.0.0",
    "@types/yup": "0.29.13",
    "just-debounce-it": "1.5.0",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "yup": "0.32.11"
  },
  "files": [
    "dist"
  ],
  "funding": [
    {
      "type": "individual",
      "url": "https://opencollective.com/formik"
    }
  ],
  "homepage": "https://formik.org",
  "jest": {
    "collectCoverageFrom": [
      "src/**/*.{ts,tsx}"
    ],
    "globals": {
      "__DEV__": "boolean"
    },
    "setupFilesAfterEnv": [
      "<rootDir>/test/setupTests.ts"
    ]
  },
  "keywords": [
    "formik",
    "form",
    "forms",
    "react",
    "react-dom",
    "hooks",
    "react hooks",
    "validation",
    "render props",
    "validation",
    "higher order component",
    "hoc"
  ],
  "license": "Apache-2.0",
  "main": "dist/index.js",
  "module": "dist/formik.esm.js",
  "name": "@gemunion/formik",
  "peerDependencies": {
    "react": ">=16.8.0"
  },
  "repository": "formium/formik",
  "scripts": {
    "build": "tsdx build --tsconfig tsconfig.build.json",
    "lint": "tsdx lint",
    "prepublish": "npm run build",
    "start": "tsdx watch --tsconfig tsconfig.build.json --verbose --noClean",
    "test": "tsdx test --env=jsdom",
    "test:watch": "npm run test -- --watchAll"
  },
  "typings": "dist/index.d.ts",
  "umd:main": "dist/formik.umd.production.js",
  "version": "2.2.9"
}