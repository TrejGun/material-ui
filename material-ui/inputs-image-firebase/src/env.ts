// eslint-disable-next-line import/no-extraneous-dependencies
import { config } from "dotenv";

config({
  path: `.env.${process.env.NODE_ENV}`,
  debug: process.env.DEBUG as any as boolean,
});

declare global {
  namespace NodeJS {
    interface ProcessEnv {
      NODE_ENV: "development" | "production" | "test" | "staging";
      FIREBASE_API_KEY: string;
      FIREBASE_AUTH_DOMAIN: string;
      FIREBASE_DB_URL: string;
      FIREBASE_PROJECT_ID: string;
      FIREBASE_STORAGE_BUCKET: string;
      FIREBASE_MESSAGE_SENDER_ID: string;
      FIREBASE_APP_ID: string;
      FIREBASE_MEASUREMENT_ID: string;
    }
  }
}
