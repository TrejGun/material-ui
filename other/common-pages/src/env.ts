declare global {
  namespace NodeJS {
    // eslint-disable-next-line @typescript-eslint/naming-convention
    interface ProcessEnv {
      BE_URL: string;
    }
  }
}

export {};
