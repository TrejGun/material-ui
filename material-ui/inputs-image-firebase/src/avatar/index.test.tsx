import "../env";
import React from "react";
import { IntlProvider } from "react-intl";
import { cleanup, render } from "@testing-library/react";
import { createTheme, ThemeProvider } from "@material-ui/core";
import { Formik } from "formik";
import { SnackbarProvider } from "notistack";

import { AvatarInput } from "./index";

afterEach(cleanup);

const i18n = {
  "form.labels.avatar": "Avatar",
  "form.tips.delete": "Delete",
  "form.validations.valueMissing": "Avatar is required",
  "form.validations.whitelistValidation": "Property Avatar is not recognized",
};

describe("<AvatarInput />", () => {
  it("renders the empty field", () => {
    const props = {
      name: "avatar",
    };

    const formikProps = {
      onSubmit: jest.fn(),
      onChange: jest.fn(),
      initialValues: {
        avatar: "",
      },
    };

    const { asFragment } = render(
      <ThemeProvider theme={createTheme()}>
        <IntlProvider locale="en" messages={i18n}>
          <SnackbarProvider maxSnack={3}>
            <Formik {...formikProps}>
              <AvatarInput {...props} />
            </Formik>
          </SnackbarProvider>
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the field with image", () => {
    const props = {
      name: "avatar",
    };

    const formikProps = {
      onSubmit: jest.fn(),
      onChange: jest.fn(),
      initialValues: {
        avatar: "https://lms2-dev.s3-us-west-2.amazonaws.com/7f0f427f-eeba-4ffb-a1c8-f730721bfd46.jpeg",
      },
    };

    const { asFragment } = render(
      <ThemeProvider theme={createTheme()}>
        <IntlProvider locale="en" messages={i18n}>
          <SnackbarProvider maxSnack={3}>
            <Formik {...formikProps}>
              <AvatarInput {...props} />
            </Formik>
          </SnackbarProvider>
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders image with error from server", () => {
    const props = {
      name: "avatar",
    };

    const formikProps = {
      onSubmit: jest.fn(),
      onChange: jest.fn(),
      initialValues: {
        avatar: "https://lms2-dev.s3-us-west-2.amazonaws.com/7f0f427f-eeba-4ffb-a1c8-f730721bfd46.jpeg",
      },
      initialErrors: {
        avatar: "form.validations.whitelistValidation",
      },
    };

    const { asFragment } = render(
      <ThemeProvider theme={createTheme()}>
        <IntlProvider locale="en" messages={i18n}>
          <SnackbarProvider maxSnack={3}>
            <Formik {...formikProps}>
              <AvatarInput {...props} />
            </Formik>
          </SnackbarProvider>
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders empty field with error: image is required", () => {
    const props = {
      name: "avatar",
    };

    const formikProps = {
      onSubmit: jest.fn(),
      onChange: jest.fn(),
      initialValues: {},
      initialErrors: {
        avatar: "form.validations.valueMissing",
      },
      initialTouched: {
        avatar: true,
      },
    };

    const { asFragment } = render(
      <ThemeProvider theme={createTheme()}>
        <IntlProvider locale="en" messages={i18n}>
          <SnackbarProvider maxSnack={3}>
            <Formik {...formikProps}>
              <AvatarInput {...props} />
            </Formik>
          </SnackbarProvider>
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
