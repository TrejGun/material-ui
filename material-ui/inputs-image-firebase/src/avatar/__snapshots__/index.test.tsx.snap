// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvatarInput /> renders empty field with error: image is required 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth AvatarInput-root-15 css-q8hpuo-MuiFormControl-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-d6jbqa-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      id="avatar-select-label"
    >
      Avatar
    </label>
    <div
      class="MuiGrid-root MuiGrid-container AvatarInput-container-16 css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="InputFile-placeholder-19"
          tabindex="0"
        >
          <input
            accept="image/jpeg,image/png"
            autocomplete="off"
            multiple=""
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium InputFile-icon-20 css-i4bv87-MuiSvgIcon-root"
            data-testid="CloudUploadIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
            />
          </svg>
        </div>
        <p
          class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          id="avatar-helper-text"
        >
          Avatar is required
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<AvatarInput /> renders image with error from server 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth AvatarInput-root-11 css-q8hpuo-MuiFormControl-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-d6jbqa-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      id="avatar-select-label"
    >
      Avatar
    </label>
    <button
      aria-label="delete"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium AvatarInput-button-13 css-1i8snpa-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element="true"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="DeleteIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <img
      alt="Avatar"
      class="AvatarInput-image-14"
      src="https://lms2-dev.s3-us-west-2.amazonaws.com/7f0f427f-eeba-4ffb-a1c8-f730721bfd46.jpeg"
    />
    <p
      class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
      id="avatar-helper-text"
    >
      Property Avatar is not recognized
    </p>
  </div>
</DocumentFragment>
`;

exports[`<AvatarInput /> renders the empty field 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth AvatarInput-root-1 css-q8hpuo-MuiFormControl-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-d6jbqa-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      id="avatar-select-label"
    >
      Avatar
    </label>
    <div
      class="MuiGrid-root MuiGrid-container AvatarInput-container-2 css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="InputFile-placeholder-5"
          tabindex="0"
        >
          <input
            accept="image/jpeg,image/png"
            autocomplete="off"
            multiple=""
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium InputFile-icon-6 css-i4bv87-MuiSvgIcon-root"
            data-testid="CloudUploadIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<AvatarInput /> renders the field with image 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth AvatarInput-root-7 css-q8hpuo-MuiFormControl-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-d6jbqa-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      id="avatar-select-label"
    >
      Avatar
    </label>
    <button
      aria-label="delete"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium AvatarInput-button-9 css-1i8snpa-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element="true"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="DeleteIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <img
      alt="Avatar"
      class="AvatarInput-image-10"
      src="https://lms2-dev.s3-us-west-2.amazonaws.com/7f0f427f-eeba-4ffb-a1c8-f730721bfd46.jpeg"
    />
  </div>
</DocumentFragment>
`;
